src/Pim/Component/Catalog/Updater/VariantGroupUpdater.php

```diff
@@ -255,104 +251,61 @@ public function setValues(GroupInterface $variantGroup, array $newValues)
-     *          },
-     *          {
-     *              "locale": "en_US",
-     *              "scope": "ecommerce",
-     *              "data": "new descriptionen_US",
-     *          }
-     *      ]
+     * New values respect the standard format:
      *
-     * Then $mergedValues will be =
-     *     "description": [
-     *          {
-     *              "locale": "fr_FR",
-     *              "scope": "ecommerce",
-     *              "data": "original description fr_FR",
-     *          },
-     *          {
-     *              "locale": "de_DE",
-     *              "scope": "ecommerce",
-     *              "data": "new description de_DE",
-     *          },
-     *          {
-     *              "locale": "en_US",
-     *              "scope": "ecommerce",
-     *              "data": "new descriptionen_US",
-     *          }
-     *      ]
+     * $newValues = [
+     *     'code'        => [
+     *         [
+     *             'locale' => null,
+     *             'scope'  => null,
+     *             'data'   => 'a_unique_code',
+     *         ],
+     *     ],
+     *     'description' => [
+     *         [
+     *             'locale' => 'en_US',
+     *             'scope'  => 'ecommerce',
+     *             'data'   => 'A new description in english',
+     *         ],
+     *         [
+     *             'locale' => 'de_DE',
+     *             'scope'  => 'ecommerce',
+     *             'data'   => 'Eine neue deutsche Beschreibung',
+     *         ],
+     *     ]
+     * ];
      *
-     * @param array $originalValues
-     * @param array $newValues
+     * @param ProductValueCollectionInterface $values
+     * @param array                           $newValues
      *
-     * @return array
+     * @return ProductValueCollectionInterface
      */
-    protected function mergeValues(array $originalValues, array $newValues)
+    protected function mergeValues(ProductValueCollectionInterface $values, array $newValues)
     {
-        $mergedValues = $originalValues;
+        foreach ($newValues as $attributeCode => $newValueArray) {
+            $attribute = $this->attributeRepository->findOneByIdentifier($attributeCode);
 
-        foreach ($newValues as $newValueCode => $newValueArray) {
             foreach ($newValueArray as $newValue) {
-                $originalValueFound = false;
-
-                if (isset($originalValues[$newValueCode])) {
-                    foreach ($originalValues[$newValueCode] as $originalValueIndex => $originalValue) {
-                        if ($newValue['locale'] === $originalValue['locale'] &&
-                            $newValue['scope'] === $originalValue['scope']
-                        ) {
-                            $originalValueFound = true;
-                            $mergedValues[$newValueCode][$originalValueIndex]['data'] = $newValue['data'];
-                        }
-                    }
-                }
+                $key = sprintf(
```

Comment:
don't do that please
the storage format should never be exposed
use $values->getByCode